client/packages/system/src/Manage/Preferences/EditPage/getPrefKey.tsx

```diff
@@ -0,0 +1,30 @@
+import { PreferencesNode } from '@common/types';
+
+export function getPrefKey(key: string): ClientPrefKey | undefined {
+  // Typescript error if any keys are missing
+  const noMissingKeys: EnsureNoMissingKeys = true;
```

Comment:
This was important to me - there is a mapping below between the pref identifier ("ServerKey") and what the object key is for that pref when served to the frontend. When we add a new pref, I wanted typescript to error until we have handled the new pref type

Reply:
100% agree, but I think I'm starting to see an issue with how we're doing this -- namely, what is the point of a query to "getAvailablePreferences", if all prefs are hard-coded in the front-end anyway?

The problem is we wanted to have a system where we could define new prefs in the back end and the front-end would display them all dynamically, but we're not really doing that any more -- so we don't really need a query to fetch them.

I think that, along with my other comment about fetching and updating one at a time the shape of the GraphQL endpoints could do with a bit of a rethink.

I have a few thoughts, but I won't go into it here. I'll try and get a few ideas down and we can have a chat about it later?

Reply:
Now handled in backend, with enum key types 🙏